osie-runner: Lets go back to split runtime/buildtime apk calls

mmlb opened the pull request:
## Description

Going back to making use of `apk`'s `virtual` package support to separate build-time vs run-time deps. Its just neater this way (DRY).

## How Has This Been Tested?

Ran `docker build`.

mmlb committed:
osie-runner: Remove stale comment

The latest version being referenced here was of pip since we were installing
from pypi.org. This is no longer the case so it should have been removed before.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
osie-runner: Minor white space fixes

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
osie-runner: Drop unneeded `apk update`

The alpine cache directory does not exist which means `apk update`
does not store any files, making this command a no-op. The cache
update happens in mem in the `apk add --update` call.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
osie-runner: Split apk calls back into runtime & buildtime

This just seems cleaner to make use of apk's --virtual packages support.
No need to remember to add a package to the apk add and apk del calls.
Case in point, osie-runner doesn't need curl at runtime but wasn't added to the
apk del call.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
osie-runner: Put && at eol instead of start

This way the commands being run all line up on the same column nicely
and look like the more common way to break up long commands into multiple
lines. Mostly for the column alignment though tbh :D.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

andy-v-h commented on osie-runner/Dockerfile:
```diff
@@ -6,22 +6,28 @@ VOLUME /statedir
+        docker \
+        jq \
+        libstdc++ \
+        python3 \
```
The base image is python:3.9, why reinstall it again?

mmlb replied:
copy/paste error, let me fix

mmlb replied:
fixed, ptal again.

mmlb committed:
osie-runner: Move COPY of requirements.txt closer to where its used

Mostly for better code organization, but also to make better use of
docker layer caching.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb committed:
osie-runner: Drop curl from build-deps

Was used to fetch the pip install script, but is no longer necessary.

Signed-off-by: Manuel Mendez <mmendez@equinix.com>

mmlb commented:
@andy-v-h gentle ping for re-review.

andy-v-h commented:
I don't think I'm an approver on this repo

mmlb commented:
> I don't think I'm an approver on this repo

Nope you are not :(
